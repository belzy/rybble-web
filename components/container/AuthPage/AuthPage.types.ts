import React from 'react';

export type TInnerContainerProps = {
  style?: Object
  bgColor?: string
  children?: React.ReactNode
};

export type TInnerContainerRightProps = {
  children?: React.ReactNode
};

