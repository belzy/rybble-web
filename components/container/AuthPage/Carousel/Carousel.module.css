
.carousel {
  min-height: calc(145px + 1rem);
}

.carousel [data-bs-target] {
  height: 10px;
  width: 10px;
  border-radius: 50%;
  background-color: #7e57c2;
  transition: all 0.3s ease-in-out;
  margin: 0 8px 0 8px;
}

.carousel [data-bs-target][class='active'] {

  height: 12px;
  width: 12px;

}

.carousel [class='carousel-indicators'] {
  align-items: center;
}

.carousel a {
  display: none;
}