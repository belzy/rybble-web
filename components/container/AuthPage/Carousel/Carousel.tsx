import React from 'react';
import { Text, } from '@chakra-ui/react';
import { 
  Carousel as BSCarousel,
} from 'react-bootstrap';
import styles from './Carousel.module.css';

const Carousel = () => {

  return (
    <BSCarousel className={ styles.carousel }>
      <BSCarousel.Item>
        <Text fontSize='3xl' as='h3' mb='16px'>Get More Done</Text>
        <Text fontSize='md'>
          Stay organised, get notified when work happens<br /> and get results.
        </Text>

      </BSCarousel.Item>

      <BSCarousel.Item>
        <Text  fontSize='3xl' as='h3' mb='16px'>Stay Connected</Text>
        <Text fontSize='md'>
          Create tasks, lists and projects and assign<br /> them to your colleagues.
        </Text>

      </BSCarousel.Item>

      <BSCarousel.Item>
        <Text fontSize='3xl' as='h3' mb='16px'>Live Smarter</Text>
        <Text fontSize='md'>
          Each minute is a little thing, and yet,  to manage<br /> the minute 
          is the secret of success.
        </Text>

      </BSCarousel.Item>
    </BSCarousel>
  );
};

export default Carousel;