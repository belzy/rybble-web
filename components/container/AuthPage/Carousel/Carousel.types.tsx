
export type TCarouselItem = {
  title: string,
  subTitle: [string, string]
};
