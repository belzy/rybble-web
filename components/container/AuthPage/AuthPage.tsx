import React from 'react';
import { 
  Flex, 
  Image, 
  Link,
  Heading,
} from '@chakra-ui/react';
import Carousel from './Carousel/Carousel';
import 'bootstrap/dist/css/bootstrap.min.css';

type TInnerContainerProps = {
  style?: Object
  bgColor?: string
  children?: React.ReactChild | Array<React.ReactChild>
};

export const InnerContainer = ({ style = {}, bgColor = 'transparent', children }: TInnerContainerProps): React.ReactElement => {

  return (
    <Flex
      { ...style }
      flexDir='column'
      flex='1'
      p='64px 80px'
      bgColor={ bgColor }
    >{ children }</Flex>
  );
};

const AuthPage = () => {

  return (
    <Flex h='100vh' w='100vw'>

      <InnerContainer bgColor='blackAlpha.200' style={{ maxWidth: '720px' }}>
        <Image mb='24px' boxSize='32px' src='/assets/img/rybble-logo32x32.svg' />
        <Carousel />
        <Image src='/assets/img/auth-page-hero560x400.png' mt='80px' />

        <Flex justifyContent='center' mt='auto'>
          <Link mr='18px'><Heading textDecor='none' size='xs' color='#673AB7'>PRIVACY POLICY</Heading></Link>
          <Link ml='18px'><Heading textDecor='none' size='xs' color='#673AB7'>TERMS OF SERVICE</Heading></Link>
        </Flex>
      </InnerContainer>

      <InnerContainer bgColor='white' style={{ width: '100%' }}></InnerContainer>

    </Flex>
  );

};

export default AuthPage;