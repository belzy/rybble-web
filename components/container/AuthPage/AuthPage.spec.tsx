import { render } from '@testing-library/react';
import AuthPage, { InnerContainer } from './AuthPage';
import Carousel from './Carousel/Carousel';
import '@testing-library/jest-dom';

describe('components/AuthPage', () => {

  it('should render', () => {

    const { container } = render(<AuthPage />);

    expect(container).toMatchSnapshot();

  });

});

describe('components/AuthPage/InnerContainer', () => {

  it('should render', () => {

    const { container } = render(
      <InnerContainer>
        <h1>Snapshot Test</h1>
      </InnerContainer>);

      expect(container).toMatchSnapshot();

  });

  it('should render a heading as a child', () => {

    render(
      <InnerContainer>
        <h1>Snapshot Test</h1>
      </InnerContainer>);

    

  });

});

describe('components/AuthPage/Carousel', () => {

  it('should render', () => {

    // const { container } = render(<Carousel />);

    // expect(container).toMatchSnapshot();

  });

});