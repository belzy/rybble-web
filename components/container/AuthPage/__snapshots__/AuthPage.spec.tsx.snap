// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/AuthPage should render 1`] = `
<div>
  <div
    class="css-qy22cp"
  >
    <div
      class="css-wj7vy8"
    >
      <img
        class="chakra-image css-6qqipc"
        src="/assets/img/rybble-logo32x32.svg"
      />
      <div
        class="carousel carousel slide"
      >
        <div
          class="carousel-indicators"
        >
          <button
            aria-current="true"
            aria-label="Slide 1"
            class="active"
            data-bs-target=""
            type="button"
          />
          <button
            aria-current="false"
            aria-label="Slide 2"
            data-bs-target=""
            type="button"
          />
          <button
            aria-current="false"
            aria-label="Slide 3"
            data-bs-target=""
            type="button"
          />
        </div>
        <div
          class="carousel-inner"
        >
          <div
            class="active carousel-item"
          >
            <h3
              class="chakra-text css-1tz2i1u"
            >
              Get More Done
            </h3>
            <p
              class="chakra-text css-504jus"
            >
              Stay organised, get notified when work happens
              <br />
               and get results.
            </p>
          </div>
          <div
            class="carousel-item"
          >
            <h3
              class="chakra-text css-1tz2i1u"
            >
              Stay Connected
            </h3>
            <p
              class="chakra-text css-504jus"
            >
              Create tasks, lists and projects and assign
              <br />
               them to your colleagues.
            </p>
          </div>
          <div
            class="carousel-item"
          >
            <h3
              class="chakra-text css-1tz2i1u"
            >
              Live Smarter
            </h3>
            <p
              class="chakra-text css-504jus"
            >
              Each minute is a little thing, and yet,  to manage
              <br />
               the minute is the secret of success.
            </p>
          </div>
        </div>
        <a
          class="carousel-control-prev"
          href="#"
          role="button"
          tabindex="0"
        >
          <span
            aria-hidden="true"
            class="carousel-control-prev-icon"
          />
          <span
            class="visually-hidden"
          >
            Previous
          </span>
        </a>
        <a
          class="carousel-control-next"
          href="#"
          role="button"
          tabindex="0"
        >
          <span
            aria-hidden="true"
            class="carousel-control-next-icon"
          />
          <span
            class="visually-hidden"
          >
            Next
          </span>
        </a>
      </div>
      <img
        class="chakra-image css-ib9iq6"
        src="/assets/img/auth-page-hero560x400.png"
      />
      <div
        class="css-78bdv5"
      >
        <a
          class="chakra-link css-x0sdsu"
        >
          <h2
            class="chakra-heading css-z4ah3"
          >
            PRIVACY POLICY
          </h2>
        </a>
        <a
          class="chakra-link css-1t7utft"
        >
          <h2
            class="chakra-heading css-z4ah3"
          >
            TERMS OF SERVICE
          </h2>
        </a>
      </div>
    </div>
    <div
      class="css-1fjsovy"
    />
  </div>
</div>
`;

exports[`components/AuthPage/InnerContainer should render 1`] = `
<div>
  <div
    class="css-a77fvk"
  >
    <h1>
      Snapshot Test
    </h1>
  </div>
</div>
`;
